src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpForwardCancellationTokenToInvocationsFixer.TypeNameVisitor.cs

```diff
@@ -0,0 +1,233 @@
+
+                return GenericName(
+                    ToIdentifierToken(symbol.Name),
+                    TypeArgumentList(SeparatedList(typeArguments)));
+            }
+
+            private static QualifiedNameSyntax CreateSystemObject()
+            {
+                return QualifiedName(
+                    AliasQualifiedName(
+                        CreateGlobalIdentifier(),
+                        IdentifierName("System")),
+                    IdentifierName("Object"));
+            }
+
+            private static TTypeSyntax AddInformationTo<TTypeSyntax>(TTypeSyntax syntax)
+                where TTypeSyntax : TypeSyntax
+            {
+                syntax = syntax.WithLeadingTrivia(ElasticMarker).WithTrailingTrivia(ElasticMarker);
+                return syntax;
+            }
+
+            /// <summary>
+            /// We always unilaterally add "global::" to all named types/namespaces.  This
+            /// will then be trimmed off if possible by the simplifier
+            /// </summary>
+            private static TypeSyntax AddGlobalAlias(SimpleNameSyntax syntax)
+            {
+                return AddInformationTo(AliasQualifiedName(CreateGlobalIdentifier(), syntax));
+            }
+
+            private static IdentifierNameSyntax ToIdentifierName(string identifier)
+                => IdentifierName(ToIdentifierToken(identifier));
+
+            private static IdentifierNameSyntax CreateGlobalIdentifier()
+                => IdentifierName(Token(SyntaxKind.GlobalKeyword));
+
+            private static bool TryCreateNativeIntegerType(INamedTypeSymbol symbol, [NotNullWhen(true)] out TypeSyntax? syntax)
+            {
+                if (symbol.IsNativeIntegerType())
+                {
+                    syntax = IdentifierName(symbol.SpecialType == SpecialType.System_IntPtr ? "nint" : "nuint");
+                    return true;
+                }
+
+                syntax = null;
+                return false;
+            }
+
+            private static SyntaxToken ToIdentifierToken(string identifier)
+            {
+                var escaped = EscapeIdentifier(identifier);
+
+                if (escaped.Length == 0 || escaped[0] != '@')
+                {
+                    return Identifier(escaped);
+                }
+
+                var unescaped = identifier.StartsWith("@", StringComparison.Ordinal)
+                    ? identifier[1..]
+                    : identifier;
+
+                var token = Identifier(
+                    default, SyntaxKind.None, "@" + unescaped, unescaped, default);
+
+                if (!identifier.StartsWith("@", StringComparison.Ordinal))
+                {
+                    token = token.WithAdditionalAnnotations(Simplifier.Annotation);
+                }
+
+                return token;
+            }
+
+            private static string EscapeIdentifier(string identifier)
```

Comment:
Should this go in `SyntaxFacts`?

Reply:
we don't have a `SyntaxFacts` type that is useable in analyzers for this repo. not a bad thing to add in a subsequent PR